java-sdk/connectors/random-connector/src/main/java/com/lucidworks/fusion/connector/plugin/RandomContentFetcher.java

```diff
@@ -58,19 +60,29 @@ public FetchResult fetch(FetchContext fetchContext) {
     FetchInput input = fetchContext.getFetchInput();
     logger.info("Received FetchInput -> {}", input);
     String hostname = getHostname();
-    long num = (Long) input.getMetadata().get("number");
-    String headline = generator.makeSentence(true);
-    int numSentences = getRandomNumberInRange(10, 255);
-    String txt = generator.makeText(numSentences);
-    logger.info("Emitting Document -> number {}", num);
-
-    Map<String, Object> fields = new HashMap();
-    fields.put("number_i", num);
-    fields.put("timestamp_l", Instant.now().toEpochMilli());
-    fields.put("headline_s", headline);
-    fields.put("hostname_s", hostname);
-    fields.put("text_t", txt);
-    fetchContext.emitDocument(fields);
+
+    try {
+      long num = (Long) input.getMetadata().get("number");
+
+      String headline = generator.makeSentence(true);
+      int numSentences = getRandomNumberInRange(10, 255);
+      String txt = generator.makeText(numSentences);
+      logger.info("Emitting Document -> number {}", num);
+
+      Map<String, Object> fields = new HashMap();
+      fields.put("number_i", num);
+      fields.put("timestamp_l", Instant.now().toEpochMilli());
+      fields.put("headline_s", headline);
+      fields.put("hostname_s", hostname);
+      fields.put("text_t", txt);
+      fetchContext.emitDocument(fields);
+    } catch (NullPointerException npe) {
+      if (ERROR_ID.equals(input.getId())) {
+        logger.info("The following error is expected, as means to demonstrate how errors are emitted");
```

Comment:
I believe that the exception being thrown was what caused the runtime framework to know there was an error, which would cause it to then emit an Error back to the server - with the changes you have here, I think we'll end up with a Candidate instead of Error in the CrawlDB. Can you do a before/after check to confirm that assumption?

Reply:
you're right, fixing this.